packages/material-ui-lab/src/PickersDay/PickersDay.tsx

```diff
@@ -2,82 +2,130 @@ import * as React from 'react';
+};
+
+const PickersDayRoot = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'MuiPickersDay',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return {
+        ...styles.root,
+        ...(!styleProps?.disableMargin && styles.dayWithMargin),
+        ...(!styleProps?.disableHighlightToday && styleProps?.today && styles.today),
+        ...(!styleProps?.outsideCurrentMonth &&
+          styleProps?.showDaysOutsideCurrentMonth &&
+          styles.dayOutsideMonth),
+      };
     },
-    '&$disabled': {
-      color: theme.palette.text.secondary,
+  },
+)(({ theme, styleProps }) => ({
+  ...(theme.typography.caption as React.CSSProperties),
+  width: DAY_SIZE,
+  height: DAY_SIZE,
+  borderRadius: '50%',
+  padding: 0,
+  // background required here to prevent collides with the other days when animating with transition group
+  backgroundColor: theme.palette.background.paper,
+  color: theme.palette.text.primary,
+  '&:hover': {
+    backgroundColor: alpha(theme.palette.action.active, theme.palette.action.hoverOpacity),
+  },
+  '&:focus': {
+    backgroundColor: alpha(theme.palette.action.active, theme.palette.action.hoverOpacity),
+    [`&.${pickersDayClasses.selected}`]: {
+      willChange: 'background-color',
+      backgroundColor: theme.palette.primary.dark,
     },
   },
-  dayWithMargin: {
-    margin: `0 ${DAY_MARGIN}px`,
+  [`&.${pickersDayClasses.selected}`]: {
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.primary.main,
+    fontWeight: theme.typography.fontWeightMedium,
+    transition: theme.transitions.create('background-color', {
+      duration: theme.transitions.duration.short,
+    }),
+    '&:hover': {
+      willChange: 'background-color',
+      backgroundColor: theme.palette.primary.dark,
+    },
   },
-  dayOutsideMonth: {
+  [`&.${pickersDayClasses.disabled}`]: {
     color: theme.palette.text.secondary,
   },
-  hiddenDaySpacingFiller: {
-    visibility: 'hidden',
-  },
-  today: {
-    '&:not($selected)': {
-      border: `1px solid ${theme.palette.text.secondary}`,
-    },
+  ...(!styleProps?.disableMargin && {
+    margin: `0 ${DAY_MARGIN}px`,
+  }),
+  ...(styleProps?.outsideCurrentMonth &&
+    styleProps?.showDaysOutsideCurrentMonth && {
+      color: theme.palette.text.secondary,
+    }),
+  ...(!styleProps?.disableHighlightToday &&
+    styleProps?.today && {
+      [`&:not(.${pickersDayClasses.selected})`]: {
+        border: `1px solid ${theme.palette.text.secondary}`,
+      },
+    }),
+}));
+
+const PickersDayFiller = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiPickersDay',
+    slot: 'Filler',
+    overridesResolver: (props, styles) => styles.hiddenDaySpacingFiller,
   },
-  selected: {},
-  disabled: {},
-});
+)(({ styleProps }) => ({
+  width: DAY_SIZE,
+  height: DAY_SIZE,
+  visibility: 'hidden',
+  ...(!styleProps?.disableMargin && {
```

Comment:
Shouldn't this component contain all styles as the other `Root` + the `visibility: 'hidden'`? I know some may not have effect, but it is easier to reason about. For example, we are missing now the `borderRadius`. I'd suggest extracting the styles and reusing them in both `styled()` components. Also, we can use the `slot: 'Root'` here too. Check also the `overridesResolver`, I believe it should be updated too to contain all styles as the `root` + the `hiddenDaySpacingFiller`.

Reply:
okay, that make sense to keep the same logic before migration. will update soon

Reply:
fixed